### ISSUE TITLE
RChain community building office hours: Story Telling and Test Cases

### ISSUE THREAD

**Issue by dckc** [rchain/bounties#403]
Let's tell stories and do mentoring and produce testable results -- and have a little fun and maybe jam while we're at it. Feel free to drop by **Saturdays, 9am Kansas City time** ( [RChain Community Calendar](https://calendar.google.com/calendar/embed?src=2cj152c9nidh6glpr1d5g4eq28%40group.calendar.google.com&ctz=America%2FChicago)).

## Recent sessions
 - [30 Mar 2019 video](https://www.youtube.com/watch?v=M4KW8imeGU4)
 - [23 Mar 2019 video](https://www.youtube.com/watch?v=9_ewngjX0Do)

[RChain Collaboration slides](https://docs.google.com/presentation/d/1KCO48W6e2voHUapr7ZB0xASnPcogeFxnNrGbFcJwHK4/edit#slide=id.p)

## Sep 29

  - venue: https://zoom.us/j/6853551826 (@jimscarver 's zoom room)
  - [shared notes](https://docs.google.com/document/d/1aGR-0dRsp2IuSAHh-opo5SAKgVJos1FCeY0V8rEy9s8/edit) _somewhat transitory; important stuff gets moved to relevant issues or the like_

## Previously
  - Sep 22: cancelled

  - https://github.com/JoshOrndorff/Rchain-Status
  - rholang discord bot

  - Sep 1: canceled in favor of RCon3 travel prep

  - Aug 25: Rchain-Status, rholang bot
  - Aug 18: cross-shard testing
  - Aug 11: RChain-API presentation; responsive theme for rewards.rchain.coop
  - Aug 4: bounty staging; SQL -> RChain

  - July 28: RChain-API coding
  - July 21: JS / RNode (#561 )
  - July 14: rewards-test staging (stuck at discord integration); node testing (rnode-bob...) #830 
  - July 7 - cancelled; holiday

  - Jun 30
    - venue: https://zoom.us/j/112294275 (host: ??? )
      - can't record without the host there
    - Singapore bizdev sync-up w/@ottermagically
    - @jimscarver on
      - user-centered design for bounty system #789 
      - size of trust metric community #785 
  - Jun 23: no takers. @jimscarver missed it; sent me a calendar invitation for Jun 30
  - June 9, 16: Cancelled. Vacation.
  - Jun 2: Rholang tutorials (phone example, rock scissors, paper, ...)

  - May 26: _IOU notes_
 - May 19
    - [potential topics](https://github.com/rchain/bounties/issues/403#issuecomment-390356565) below
  - May 12 - **Cancelled**
  - May 5
    - #375 trust metric; terms of service
    - identity stuff

  - Apr 31 (_IOU a couple words of summary_)
  - Apr 21
    - #375 trust metric; terms of service
    - #279 membership certificate (in rholang)
  - Apr 14 preempted by Dev conference
  - Apr 7 (_IOU a couple words of summary_)

  - March 31 ??
  - March 24 Agenda
    - [Social Media Strangle Hold](https://github.com/rchain/bounties/wiki/Social-Media-Strangle-Hold) anti-pattern
    - [sharecropping issues](https://github.com/rchain/bounties/labels/sharecropping) #386 #458
    - **member-site**: 4 Week UI/UX Mock Ups #446
    - Marketing-automation #219

   - fallback: [madmode hangout](https://hangouts.google.com/hangouts/_/madmode.com/dckc?authuser)

----

## Background

> Greg Meredith dedicated funds a bounty system for working together. The rewards are intended to (1) build a thriving open source developer community, and (2) involve the creative community in telling the RChain story -- [Welcome to the Bounty Program][1]

[1]: https://github.com/rchain/bounties/wiki/Welcome-to-the-Bounty-Program

I come to the RChain community with a couple decades of [relevant experience from W3C][yrs] and [other places in my career](https://www.linkedin.com/in/connollydan/) and I'd like to share it.

I'm also on the [RChain executive committee](https://www.gov.rchain.coop/executive-committee).

Perhaps you met me Feb 2018 in Seattle where we were [dreaming big at the RChain Governance Forum](http://www.madmode.com/2018/rchain-gov-forum-SEA.html). Or in Boulder where gave a [The Web is Under Threat. Can RChain Answer?](https://goo.gl/PAsM7q) talk and another on [Getting Involved in the RChain Bounty Program][bt]

[bt]: https://docs.google.com/presentation/d/1B2Vu8o3ACwruY6HY1ayXRQ4qkNKsMy4hdbOdxrCHI2o/edit#slide=id.p

cc @drbloom @patrick727 @kennyrowe 

### Patterns:

 - [StoryTellingAndTestCases](https://informatics.gpcnetwork.org/trac/Project/wiki/StoryTellingAndTestCases) in the wiki of the GPC, a multi-site health data research project where I'm the technical lead.
   - I have 20+ years development in python, C, scala, etc. My first open source release was in Jun 1992. I read the source code to most of the software I use (linux, X windows, browsers, ...) I'm pretty active in github.
   - I have done 100+ 1-1 mentoring sessons; my [dckc codementor profile](https://www.codementor.io/dckc) has 30+ reviews. 
 - [MidwestWeeklyAgenda](https://www.w3.org/wiki/MidwestWeeklyAgenda) - a distillation of my experience running meetings at W3C (see also [GPC issue 12](https://informatics.gpcnetwork.org/trac/Project/ticket/12)). I value integrity and I know how to engender it, using the web for accountability. I know the cost of bad meetings and what it takes to make good ones.
  - **object capability discipline** - As I said in [my Capability Security 2017 Wish-List](http://www.madmode.com/2017/ocap-wish-list.html), Computers are getting faster, smaller, more connected, and more capable, but when it comes to security, everything is broken. Along with correct-by-construction software, the best weapon I see is object capability discipline, which is part of the RChain architecture..

[yrs]: https://www.w3.org/blog/2010/06/thanks-for-a-great-15-years-at/

**Comment by Jake-Gillberg** [comment]
Color me interested!

**Comment by dckc** [comment]
@Jake-Gillberg @ICA3DaR5 do you want to start this Sat, Feb 24?

**Comment by dckc** [comment]
@Jake-Gillberg here's the (start of) the NBAC, ZBAC story...

## Capability Security

> Computers are getting faster, smaller, more connected, and more capable, but when it comes to security, everything is broken. Along with correct-by-construction software, the best weapon I see is **object capability discipline**. -- [My Capability Security 2017 Wish-List](http://www.madmode.com/2017/ocap-wish-list.html)

**Comment by dckc** [comment]
ok; let's try this... [hangout](https://hangouts.google.com/hangouts/_/madmode.com/dckc?authuser=)

**Comment by BelovedAquila** [comment]
@dckc you made a mistake here
 >  @ICA3DaR5 I hope to year your story...

**Comment by dckc** [comment]
What mistake is that, @BelovedAquila ? I don't see it.

**Comment by BelovedAquila** [comment]
@dckc you said "I hope to YEAR your story" was thinking that should be "I hope to HEAR your story" what do you see?

**Comment by dckc** [comment]
ah. thanks. fixed.

(I'm going to delete these comments now.)

**Comment by dckc** [comment]
today's location: https://zoom.us/j/899291322

**Comment by ResonanceXX** [comment]
Tried accessing the Zoom link via Android. But got an error message "Meeting link is invalid"


Sent from Yahoo Mail on Android 
 
  On Sat, 3 Mar 2018 at 15:56, Dan Connolly<notifications@github.com> wrote:   
today's location: https://zoom.us/j/899291322

**Comment by ICA3DaR5** [comment]
@dckc The link you posted returns: "Invalid meeting ID. Your meeting ID is a 9, 10, or 11-digit number"

@ResonanceXX Try downloading the Android Zoom Mobile App fro the Google Play Store: https://zoom.us/download#mobile_app

**Comment by dckc** [comment]
Yes, well, the meeting time was 9am Chicago time. By the time of your comments, it was over.

**Comment by drbloom** [comment]
I can attend March 10 7AM PST / 10AM EST and offer to host the meeting in a Zoom Room:

Join from PC, Mac, Linux, iOS or Android: [https://zoom.us/j/196762842](https://zoom.us/j/196762842)

Or iPhone one-tap :
    US: +16699006833,,196762842# or +19294362866,,196762842# 
Or Telephone:
    Dial(for higher quality, dial a number based on your current location): 
        US: +1 669 900 6833 or +1 929 436 2866 
    Meeting ID: 196 762 842
    International numbers available: https://zoom.us/zoomconference?m=g_t1lFxD73ADHOnHx_3FvgJcK-RmsbUK

**Comment by dckc** [comment]
@AbnerZheng  let's do talk about readthedocs and translations.

[Saturday, 9am my time, 10pm yours](https://www.timeanddate.com/worldclock/meetingdetails.html?year=2018&month=3&day=24&hour=14&min=0&sec=0&p1=405&p2=33)

**Comment by ian-bloom** [comment]
@dckc agenda item related to readthedocs mentioned above. I'd like to discuss "Sharecropping", hosting Governance related docs, Cool URI's, and general document store #386 
I set up a Zoom room for 3/24 [10am my time, 9am yours](https://www.timeanddate.com/worldclock/meetingdetails.html?year=2018&month=3&day=24&hour=14&min=0&sec=0&p1=405&p2=33&p3=263) :-)

[https://zoom.us/j/181846155](https://zoom.us/j/181846155)

**Comment by dckc** [comment]
I may continue to keep notes here, but there's not much reason to keep it among the open issues.

**Comment by dckc** [comment]
Regardless of the state of the bounty program, I still do this reasonably regularly.

**Comment by jimscarver** [comment]
today's topic https://github.com/rchain-community/sm19a 
video https://youtu.be/x-YrKbdglnw
